pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -57,6 +69,50 @@ func (am *Alertmanager) IsDisabled() bool {
 
 func (am *Alertmanager) Init() error {
 	am.logger = log.New("alertmanager")
+	am.Store = store.DBstore{} //TODO: Is this right?
+
+	err := am.Setup()
+	if err != nil {
+		return errors.Wrap(err, "unable to start the Alertmanager")
+	}
+
+	return nil
+}
+
+// Setup takes care of initializing the various configuration we need to run an Alertmanager.
+func (am *Alertmanager) Setup() error {
+	// First, let's get the configuration we need from the database and settings.
+	q := &models.GetLatestAlertmanagerConfigurationQuery{}
+	if err := am.Store.GetLatestAlertmanagerConfiguration(q); err != nil {
+		return err
+	}
+
+	// Next, let's parse the alertmanager configuration.
+	cfg, err := parse(q.Result.AlertmanagerConfiguration, q.Result.AlertmanagerTemplates)
+	if err != nil {
+		return err
+	}
+
+	// With that, we need to make sure we persist the templates to disk.
+	paths, _, err := cfg.PersistTemplates(am.WorkingDirPath())
+	if err != nil {
+		return err
+	}
+
+	// With the templates persisted, create the template list using the paths.
+	tmpl, err := template.FromGlobs(paths...)
+	if err != nil {
+		return err
+	}
+
+	// Finally, build the integrations map using the receiver configuration and templates.
+	am.integrationsMap, err = am.buildIntegrationsMap(cfg.Alertmanager.Receivers, tmpl)
```

Comment:
You might want to use the [new config](https://github.com/grafana/alerting-api/blob/6ff97279b2f185d1ce4a16e9c92d08d905bd0f07/pkg/api/alertmanager.go#L431
) that will be used for notification channels here.

Hence [this](https://github.com/grafana/alerting-api/blob/6ff97279b2f185d1ce4a16e9c92d08d905bd0f07/pkg/api/alertmanager.go#L244-L249) config instead of Prometheus alertmanager config

Reply:
Done!